subsys/testsuite/include/negative_handler.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+
+
+#if defined(CONFIG_NEGATIVE_TEST_FATAL_HANDLER)
+/* These valid flag used to figure out if the fault was expected or not. */
+extern volatile bool valid_fault;
```

Comment:
It may not be an issue for the specific tests using this feature, but note that this isn't SMP-safe.  A fault happens on a given CPU, calling "set_fault_valid()" will race against anything going wrong on the other CPU (which, again, may not be an issue in practice).

Really this tracking needs to be on the _current_cpu struct, not global.  But again maybe not an issue since it's only a handful of tests that need this and they can arrange things on their own to not have racing faults.

Reply:
Yes, I totally did not consider this. It indeed be able to cause problem under SMP cases. Thank you so much for your reminder on this! I will take into account this more and think about how to refine this.

Reply:
I try to give each cpu a valid_fault or valid_assert flag to make sure they can be used in SMP case. But I did not put the in _current_cpu struct. Could you please to see if my method works? thanks!